controllers/clustermanagementaddon_controller.go

```diff
@@ -487,3 +488,21 @@ func getPEMCertExpireTime(pemBytes []byte) *metav1.Time {
 	}
 	return &metav1.Time{Time: cert.NotAfter}
 }
+
+func getAnnotation(list []proxyv1alpha1.AnnotationVar) map[string]string {
+	if len(list) == 0 {
+		return nil
+	}
+	annotation := make(map[string]string, len(list))
+	for _, v := range list {
+		key := v.Key
+		if len(key) == 0 {
+			continue
+		}
+		if len(key) > 63 {
```

Comment:
either conform the label validation w/ the k8s original library i.e. `validation.IsQualifiedName` on the key and `validation.IsValidLabelValue` on the value or removed the validation to make it simpler